### PR TITLE
Update

### PR DESCRIPTION
**Pull request by Heptazhou**
+ Update CI workflow
+ Reduce redirection for links in README
+ Replace `!= nothing` with `!== nothing`
+ Fix typos

### CODE REVIEW

**Comment by Heptazhou** [comment]
actually `!= nothing` is several times slower than `!== nothing` or `!isnothing()` ...

File: src/nest_utils.jl [hunk 1]
```diff
@@ -35,7 +35,7 @@ should return a value other than `nothing`.
 """
 function walk(f, fst::FST, s::State)
     stop = f(fst, s)
-    (stop != nothing || is_leaf(fst)) && return
+    (!isnothing(stop) || is_leaf(fst)) && return
```
Comment (domluna) [inline src/nest_utils.jl:38]: I don't think this is compatible with earlier versions of Julia so we need to use !== or ===

**Review by domluna — COMMENTED** [review]
Can you add a description for the PR, it looks like it updates the CI workflow and typos, etc

**Comment by domluna** [comment]
@Heptazhou looks good, let me know if this is done

**Comment by Heptazhou** [comment]
@domluna It's good to go.

**Pull request merged**